src/compiler/compiler.ts

```diff
@@ -22,21 +22,24 @@ export function compile(spec, theme?) {
   var model = new Model(spec, theme);
   const layout = model.layout();
 
-  var rootGroup:any = {
-    name: 'root',
-    type: 'group',
-    from: {data: LAYOUT},
-    properties: {
-      update: {
-        width: layout.width.field ?
-               {field: layout.width.field} :
-               {value: layout.width},
-        height: layout.height.field ?
-                {field: layout.height.field} :
-                {value: layout.height}
+  let rootGroup:any = extend({
+      name: spec.name ? spec.name + '_root' : 'root',
+      type: 'group',
+    },
+    spec.description ? {description: spec.description} : {},
```

Comment:
I disagree.  The goal of this chunk of code is to optimize order of the vega output. 
While stylistically a little uglier, putting it like this allows you to determine order of the output in one glance.